protocol/x/vault/keeper/shares.go

```diff
@@ -127,7 +175,37 @@ func (k Keeper) MintShares(
 		return err
 	}
 
-	// TODO (TRA-170): Increase owner shares.
+	// Increase owner shares in the vault.
+	ownerShares, exists := k.GetOwnerShares(ctx, vaultId, owner)
+	if !exists {
+		// Set owner shares to be sharesToMint.
+		err := k.SetOwnerShares(
+			ctx,
+			vaultId,
+			owner,
+			types.BigRatToNumShares(sharesToMint),
+		)
+		if err != nil {
+			return err
+		}
+	} else {
+		// Increase existing owner shares by sharesToMint.
+		existingOwnerShares, err := ownerShares.ToBigRat()
```

Comment:
to confirm, there are no race conditions here because everything in abci is synchronous right?